packages/msw-addon/src/mswDecorator.js

```diff
@@ -1,23 +1,41 @@
-import { makeDecorator } from '@storybook/addons';
 import { setupWorker } from 'msw';
+import { setupServer } from 'msw/node';
 
 let worker;
 
+const IS_BROWSER = typeof global.process === 'undefined'
+
 export function initializeWorker(options) {
-  if (typeof global.process === 'undefined') {
+  if (IS_BROWSER) {
     worker = setupWorker();
     worker.start(options);
+  } else {
+    worker = setupServer();
+    worker.listen(options);
+  }
+}
+
+export function getWorker() {
+  if(worker === undefined) {
+    throw new Error(`[MSW] Tried to get worker but it was not defined yet. Did you forget to initialize it?`)
   }
+
+  return worker;
 }
 
-export const mswDecorator = makeDecorator({
-  name: 'withMsw',
-  parameterName: 'msw',
-  wrapper: (storyFn, context, { parameters = [] }) => {
-    if (worker) {
-      worker.resetHandlers();
-      worker.use(...parameters);
+export const mswDecorator = (storyFn, { parameters: { msw = [] } }) => {
+  if (worker) {
+    worker.resetHandlers();
+
+    if (Array.isArray(msw)) {
+      if (msw.length > 0) {
+        worker.use(...msw);
+      }
+    } else {
+      throw new Error(`[MSW] expected to receive an array of handlers but received "${typeof msw}" instead.
```

Comment:
A nitpick: we can check for the error scenario first, allowing that very check to be more evident, and operating with the `msw` variable assured it an array for the rest of the function:

```js
if (!Array.isArray(msw)) {
  throw new Error(...)
}

// No need to "else", msw is an array from now on.
```

I tend to find explicit early checks for error scenarios simplifying the code a bit.

Reply:
Good one!